### ISSUE TITLE
Is there a plan to add performance optimization 

### ISSUE THREAD

**Issue by zhonglin6666** [longhorn/longhorn#1002]
Found longhorn write into back file，Can cache be added to improve performance ？

Is there a plan to add performance optimization ？

**Comment by yasker** [comment]
We don't have a specific plan to add cache now. Longhorn is designed to be crash-consistent. Any cache (especially cache for writing small IOs) will make it no longer crash-consistent since cache can be lost. So if the disk was cut off from power suddenly, the user might lose some part of the data that already been considered written in the cache. It can be a few ms to seconds worth of the data.

We do plan to look into ways to improve performance. We can keep this issue open to track any future effort on improving the performance using cache.

**Comment by michaelandrepearce** [comment]
@yasker cache could improve read io without impacting crash consistency by making writes through the cache.

this is actually a highlighted performance miss in this competitive report

https://storageos.com/wp-content/uploads/2021/02/Performance-Benchmarking-Cloud-Native-Storage-Solutions-for-Kubernetes.pdf

**Comment by yasker** [comment]
@michaelandrepearce yeah saw that too. We're expanding the team and aiming to vastly improve the performance in the next couple of minor releases.
